src/azul/queues.py

```diff
@@ -430,19 +433,26 @@ def manage_lambdas(self, queues: Mapping[str, Queue], enable: bool):
             def submit(f, *args, **kwargs):
                 futures.append(tpe.submit(f, *args, **kwargs))
 
+            indexer = load_app_module('indexer')
+            lambdas = [
+                indexer.contribute,
+                indexer.contribute_retry,
+                indexer.aggregate,
+                indexer.aggregate_retry
+            ]
+            queues_to_lambda = {lambda_.queue: lambda_ for lambda_ in lambdas}
             for queue_name, queue in queues.items():
                 if queue_name == config.notifications_queue_name():
+                    # Since the indexer lambda writes to the notifications
+                    # queue, we must deactivate it also.
                     submit(self._manage_lambda, config.indexer_name, enable)
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute', queue, enable)
-                elif queue_name == config.notifications_queue_name(retry=True):
-                    submit(self._manage_sqs_push, config.indexer_name + '-contribute_retry', queue, enable)
-                elif queue_name == config.tallies_queue_name():
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate', queue, enable)
-                elif queue_name == config.tallies_queue_name(retry=True):
-                    # FIXME: Brittle coupling between the string literal below and
-                    #        the handler function name in app.py
-                    #        https://github.com/DataBiosphere/azul/issues/1848
-                    submit(self._manage_sqs_push, config.indexer_name + '-aggregate_retry', queue, enable)
+                try:
+                    lambda_ = queues_to_lambda[queue_name]
+                except KeyError:
+                    assert 'fail' in queue_name
+                else:
+                    lambda_name = f'{config.indexer_name}-{lambda_.lambda_name}'
```

Comment:
Surely there is sth in config for this purpose?

Reply:
This is somewhat complicated, would like to discuss in parking lot.

Reply:
My idea about bypassing suffix was wrong and doesn't work. There's still not an elegant way to do this. I'd like to code through this with you @hannes-ucsc so that we can come up with a satisfying answer.